### PR TITLE
Compile with -safe-string

### PR DESCRIPTION
**Pull request by ejgallego**
This allows ssreflect to compile with coq/coq#134.

Note, this has to be merged in coordination with the Coq patch, as it enables a new API `Id.of_bytes` which is needed, so I suggest to wait until the Coq patch is given the green light.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,11 +17,14 @@ env:
   - NJOBS=2
   # system is == 4.02.3
   - COMPILER="system"
+  # - COQ_GIT=git://scm.gforge.inria.fr/coq/coq.git
+  - COQ_GIT=https://github.com/ejgallego/coq.git
```
Comment (gasche) [inline .travis.yml]: Do you really want to still point to your personal repo once your PR has been merged on the Coq side?

Comment (ejgallego) [inline .travis.yml]: No, this patch will be dropped once we agree on a merge. We have a better overlay system on the works (one that actually picks the proper overlay based on `TRAVIS_PR` etc... but it is not ready yet so we have to resort to workaround.

File: mathcomp/ssreflect/plugin/trunk/ssreflect.ml4 [hunk 2]
```diff
@@ -1238,19 +1240,20 @@ let interp_search_notation loc tag okey =
...
       let m = j - i in
       if s.[i] = '\'' && i < j - 2 && s.[j - 1] = '\'' then
         (String.blit s (i + 1) s' i' (m - 2); loop (j + 1) (i' + m - 1))
       else if for_key && is_ident (String.sub s i m) then
-         (s'.[i'] <- '_'; loop (j + 1) (i' + 2))
+         (Bytes.set s' i' '_'; loop (j + 1) (i' + 2))
       else (String.blit s i s' i' m; loop (j + 1) (i' + m + 1)) in
     loop 0 1 in
-  let trim_ntn (pntn, m) = String.sub pntn 1 (max 0 m) in
+  let trim_ntn (pntn, m) = Bytes.sub pntn 1 (max 0 m) in
```
Comment (gasche) [inline mathcomp/ssreflect/plugin/trunk/ssreflect.ml4]: Couldn't you have a simpler patch if you converted back to string before returning in `mk_pntn`? I have the impression that all the other operations below could remain on strings only (do you modify a `ntn` after it has been returned by `mk_ntn`?).

Comment (ejgallego) [inline mathcomp/ssreflect/plugin/trunk/ssreflect.ml4]: Right! Thanks again Gabriel for your invaluable feedback.

**Comment by gares** [comment]
Also @maximedenes is interested, since such code is in his ssr/quill branch

**Comment by ejgallego** [comment]
Overlay removed as this is being merged upstream, previous build at: https://travis-ci.org/ejgallego/math-comp/builds/204768299

**Comment by ejgallego** [comment]
Thanks!!!

**Pull request merged**